src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2478,7 +2482,50 @@ public ConstructorEdit(INamedTypeSymbol oldType)
                                             continue;
                                         }
 
-                                        if (!newSymbol.IsImplicitlyDeclared)
+                                        if (IsGlobalStatement(edit.OldNode))
+                                        {
+                                            // A delete of a global statement, when newSymbol isn't null, is an update to the implicit Main method
+                                            editKind = SemanticEditKind.Update;
+
+                                            // Since we're deleting we don't have any new nodes, so we have to use the declaring syntax reference 
+                                            Contract.ThrowIfTrue(oldDeclaration == null);
+                                            Contract.ThrowIfFalse(newDeclaration == null);
+                                            newDeclaration = GetSymbolDeclarationSyntax(newSymbol.DeclaringSyntaxReferences[0], cancellationToken);
+
+                                            // If all global statements are moved from one file to another then we'll get insert edits for them
+                                            // so we don't need to do anything for the deletes.
+                                            if (newDeclaration.SyntaxTree != newModel.SyntaxTree)
+                                            {
+                                                continue;
```

Comment:
>continue [](http://example.com/codeflow?start=48&length=8)

Do we get here if we started with global statements in two files and the edit removed all global statements from the first file without modifying the second file?

Reply:
Yes, we would end up here. I don't think thats an issue though, as its a compile error to have global statements in more than one file so I don't think the scenario where it could happen. If the project doesn't compile the user couldn't have started an EnC session.